fiasko_bro/code_validator.py

```diff
@@ -123,17 +128,21 @@ def _run_warning_validators_until(self, failed_error_group_name, arguments):
 
     def validate(self, repo_path, original_repo_path=None, check_repo_size=True, **kwargs):
         self.validator_arguments.update(kwargs)
+        self.validator_arguments['path_to_repo'] = repo_path
+        self.validator_arguments['original_repo_path'] = original_repo_path
         self.validator_arguments['whitelists'] = self.whitelists
         self.validator_arguments['blacklists'] = self.blacklists
-        max_num_of_py_files = self.validator_arguments['max_num_of_py_files']
-        if check_repo_size:
-            if is_repo_too_large(repo_path, max_num_of_py_files, original_repo_path):
-                return [('repo is too large', '')]
-        self.validator_arguments['solution_repo'] = LocalRepositoryInfo(
-            repo_path)
+        errors = []
+        for error_group_name, error_group in self.pre_validation_checks.items():
```

Comment:
Now, this is clearly a code duplication with the line 147. Maybe factor this out into a separate method which takes validator groups as a parameter